src/main/java/ch/njol/skript/expressions/ExprDirection.java

```diff
@@ -36,177 +28,185 @@
+	"spawn a creeper 1.5 meters horizontally behind the player",
+	"spawn a TNT 5 meters above and 2 meters horizontally behind the player",
+	"thrust the last spawned TNT in the horizontal direction of the player with speed 0.2",
+	"push the player upwards and horizontally forward at speed 0.5",
+	"push the clicked entity in in the direction of the player at speed -0.5",
+	"open the inventory of the block 2 blocks below the player to the player",
+	"teleport the clicked entity behind the player",
+	"grow a regular tree 2 meters horizontally behind the player"})
+@Since("1.0 (basic), 2.0 (extended), INSERT VERSION (pattern changes)")
 public class ExprDirection extends SimpleExpression<Direction> {
-	
-	private final static BlockFace[] byMark = new BlockFace[] {
-			BlockFace.UP, BlockFace.DOWN,
-			BlockFace.NORTH, BlockFace.SOUTH, BlockFace.EAST, BlockFace.WEST,
-			BlockFace.NORTH_EAST, BlockFace.NORTH_WEST, BlockFace.SOUTH_EAST, BlockFace.SOUTH_WEST};
-	private final static int UP = 0, DOWN = 1,
-			NORTH = 2, SOUTH = 3, EAST = 4, WEST = 5,
-			NORTH_EAST = 6, NORTH_WEST = 7, SOUTH_EAST = 8, SOUTH_WEST = 9;
-	
+
+	private static final Patterns<BlockFace> PATTERNS = new Patterns<>(new Object[][]{
+		// northeast, northwest, southeast and southwest need to be registered before normal usages otherwise `[-%direction%]` is given priority
+		// naturally doesn't matter however this should help a bit with parsing "north west north west north west..."
+		{"[%-number% [(block|meter|metre)[s]]] [to the] north[-| ]east[er(n|ly)|ward[s|ly]] [[side] of] [[and|,] %-direction%]", BlockFace.NORTH_EAST},
+		{"[%-number% [(block|meter|metre)[s]]] [to the] north[-| ]west[er(n|ly)|ward[s|ly]] [[side] of] [[and|,] %-direction%]", BlockFace.NORTH_WEST},
+		{"[%-number% [(block|meter|metre)[s]]] [to the] south[-| ]east[er(n|ly)|ward[s|ly]] [[side] of] [[and|,] %-direction%]", BlockFace.SOUTH_EAST},
+		{"[%-number% [(block|meter|metre)[s]]] [to the] south[-| ]west[er(n|ly)|ward[s|ly]] [[side] of] [[and|,] %-direction%]", BlockFace.SOUTH_WEST},
+		{"[%-number% [(block|meter|metre)[s]]] [to the] north[er(n|ly)|ward[s|ly]] [[side] of] [[and|,] %-direction%]", BlockFace.NORTH},
+		{"[%-number% [(block|meter|metre)[s]]] [to the] east[er(n|ly)|ward[s|ly]] [[side] of] [[and|,] %-direction%]", BlockFace.EAST},
+		{"[%-number% [(block|meter|metre)[s]]] [to the] south[er(n|ly)|ward[s|ly]] [[side] of] [[and|,] %-direction%]", BlockFace.SOUTH},
+		{"[%-number% [(block|meter|metre)[s]]] [to the] west[er(n|ly)|ward[s|ly]] [[side] of] [[and|,] %-direction%]", BlockFace.WEST},
+		{"[%-number% [(block|meter|metre)[s]]] (above|over|up[ward[s|ly]]) [%-direction%]", BlockFace.UP},
+		{"[%-number% [(block|meter|metre)[s]]] (below|under|down[ward[s|ly]]) [%-direction%]", BlockFace.DOWN},
+		{"[%-number% [(block|meter|metre)[s]]] [in [the]] [:horizontal] (direction|:facing) of %entity/block% [of|from]", BlockFace.SELF},
```

Comment:
Will neee to look into this or ExprFacing as a whole. Iirc, this is a pattern existing before I edited, so it existed for a while.

Optionally, if this is creating conflicts, it might be worth seeing about a merge.

I intend to add opposite facing into a few, and I know there is a pr that wants to add it to expr facing.

Could also remove pattern, but I dont think I want them in two places. Thoughts on if a merge is ideal?
Both return directions, iirc.

Reply:
The previous pattern had a required `in`